addons/s3/models.py

```diff
@@ -62,7 +62,9 @@ def folder_path(self):
     def display_name(self):
         return u'{0}: {1}'.format(self.config.full_name, self.folder_id)
 
-    def set_folder(self, folder_id, auth, bucket_name=None):
+    def set_folder(self, folder_id, auth):
+        bucket_name = folder_id.split(':')[0]
```

Comment:
double check that bucket name can not have `:`

Reply:
Bucket names can only have alphanumeric characters, plus periods and hyphens, but nothing else.